build/lib/builder.js

```diff
@@ -82,11 +133,53 @@ class Builder {
 		this.program.gitHash = hash || 'n/a';
 	}
 
+	async transpile(platform) {
+		// Copy over common dir, @babel/polyfill, etc into some temp dir
+		// Then run rollup/babel on it, then just copy the resulting bundle to our real destination!
+		// The temporary location we'll assembled the transpiled bundle
+		const TMP_COMMON_DIR = path.join(TMP_DIR, '_common');
+		const TMP_COMMON_PLAFORM_DIR = path.join(TMP_DIR, '_common', platform);
+
+		console.log(`Creating temporary 'common' directory...`); // eslint-disable-line quotes
+		await fs.copy(path.join(ROOT_DIR, 'common'), TMP_COMMON_PLAFORM_DIR);
+
+		// copy over polyfill and its dependencies
+		console.log('Copying polyfills...');
+		const modulesDir = path.join(TMP_COMMON_PLAFORM_DIR, 'Resources/node_modules');
+		// make sure our 'node_modules' directory exists
+		await fs.ensureDir(modulesDir);
+		copyPackageAndDependencies('@babel/polyfill', modulesDir);
+
+		// create a bundle
+		console.log('Transpile and run rollup...');
+		const babelOptions = determineBabelOptions(platform);
+		const bundle = await rollup({
+			input: `${TMP_COMMON_PLAFORM_DIR}/Resources/ti.main.js`,
+			plugins: [
+				resolve(),
+				commonjs(),
+				babel(babelOptions)
+			],
+			external: [ './app', 'com.appcelerator.aca' ]
+		});
+
+		console.log(`Writing 'common' bundle...`); // eslint-disable-line quotes
+		await bundle.write({ format: 'cjs', file: path.join(TMP_DIR, 'common', platform, 'ti.main.js') });
+
+		// We used to have to copy over ti.internal, but it is now bundled into ti.main.js
+		// if we ever have files there that cannot be bundled or are not hooked up properly, we'll need to copy them here manually.
+
+		console.log(`Removing temporary 'common' directory...`); // eslint-disable-line quotes
+		await fs.remove(TMP_COMMON_DIR);
+	}
+
 	async build() {
 		await this.ensureGitHash();
 		console.log('Building MobileSDK version %s, githash %s', this.program.sdkVersion, this.program.gitHash);
+
 		// TODO: build platforms in parallel
 		for (const item of this.platforms) {
+			await this.transpile(item);
```

Comment:
Here's the part I was referring to in the above comment - right below this you're constructing the Platform-specific "builder" where the logic to supply the babel `targets` value would make sense to live.